node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/WriteFileTask.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.task;
+
+import com.yahoo.vespa.hosted.node.admin.io.FileSystemPath;
+import org.glassfish.jersey.internal.util.Producer;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+public class WriteFileTask implements Task {
+    private final Params params;
+
+    public static class Params {
+        private final Path path;
+        private final Producer<String> contentProducer;
+
+        private Optional<String> user = Optional.empty();
+        private Optional<String> group = Optional.empty();
+        private Optional<String> permissions = Optional.empty();
+
+        public Params(Path path, Producer<String> contentProducer) {
+            this.path = path;
+            this.contentProducer = contentProducer;
+        }
+
+        public Params withUser(String user) {
+            this.user = Optional.of(user);
+            return this;
+        }
+
+        public Params withGroup(String group) {
+            this.group = Optional.of(group);
+            return this;
+        }
+
+        /**
+         * @param permissions of the form "rwxr-x---".
+         */
+        public Params withPermissions(String permissions) {
+            this.permissions = Optional.of(permissions);
+            return this;
+        }
+    }
+
+    public WriteFileTask(Params params) {
+        this.params = params;
+    }
+
+    @Override
+    public boolean execute(TaskContext context) {
+        final FileSystemPath path = context.getFileSystem().withPath(params.path);
+        if (path.isRegularFile()) {
```

Comment:
Should maybe compare file contents? I would expect `WriteFileTask` to overwrite an old file should the contents differ.

Reply:
Let me add a TODO for that. Some file content may be expensive to compute compared to an simpler check like existence(or timestamp, so perhaps this needs to be controlled by parameters.

Reply:
Actually there is a TODO below, but let me move it up here and rephrase.